kikuchipy/projections/lambert_projection.py

```diff
@@ -0,0 +1,115 @@
+# -*- coding: utf-8 -*-
+# Copyright 2019-2020 The kikuchipy developers
+#
+# This file is part of kikuchipy.
+#
+# kikuchipy is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# kikuchipy is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with kikuchipy. If not, see <http://www.gnu.org/licenses/>.
+
+from typing import Union
+
+import numpy as np
+from orix.vector import Vector3d
+
+from kikuchipy.projections.gnomonic_projection import GnomonicProjection
+
+
+class LambertProjection:
+    """Lambert Projection of a vector"""
+
+    @classmethod
+    def project(cls, v: Union[Vector3d, np.ndarray]) -> np.ndarray:
+        """Convert (n, 3) vector from Cartesian to the Lambert projection"""
+        if isinstance(v, Vector3d):
+            w = v.unit.data
+            x = w[..., 0]
+            y = w[..., 1]
+            z = w[..., 2]
+        else:
+            norm = np.sqrt(
+                np.sum(np.square([v[..., 0], v[..., 1], v[..., 2]]), axis=0)
+            )
+            x = v[..., 0] / norm
+            y = v[..., 1] / norm
+            z = v[..., 2] / norm
+
+        X = np.where(
+            abs(y) <= abs(x),
+            np.sign(x) * np.sqrt(2 * (1 - z)) * (sqrt_pi / 2),
+            np.sign(y)
+            * np.sqrt(2 * (1 - z))
+            * ((2 / sqrt_pi) * np.arctan2(x, y)),
+        )
+        Y = np.where(
+            abs(y) <= abs(x),
+            np.sign(x)
+            * np.sqrt(2 * (1 - z))
+            * ((2 / sqrt_pi) * np.arctan2(y, x)),
+            np.sign(y) * np.sqrt(2 * (1 - z)) * (sqrt_pi / 2),
+        )
+
+        return np.column_stack((X, Y))
+
+    @staticmethod
+    def iproject(vec: np.ndarray) -> Vector3d:
+        """Convert (n, 2) array from Lambert to Cartesian coordinates."""
+        X = vec[..., 0]
+        Y = vec[..., 1]
+
+        x = np.where(
+            abs(Y) <= abs(X),
+            _eq_c(X) * np.cos((Y * np.pi) / (4 * X)),
```

Comment:
See the comment on line 48, might be you save some time computing the cos/sin terms here only once?